src/dialect.rs

```diff
@@ -2,10 +2,26 @@ use crate::allocator::{Allocator, NodePtr};
 use crate::cost::Cost;
 use crate::reduction::Response;
 
+#[repr(u32)]
+#[derive(Clone, Copy, Eq, PartialEq)]
+pub enum Extension {
```

Comment:
I'm trying to understand why we need this new structure. I'd always figured the `Dialect` object would be the thing that changes based on extension. So you have the base dialect, which is as we see, then a dialect level 0 which adds `coinid` and the bls12 ops. Is there something that makes `Extension` necessary? Isn't it just sort of an add-on to the `Dialect`?

Reply:
in a way `Dialect` might be a pretty good name for this enum, but it's taken